src/DurableTask.Netherite/StorageLayer/Faster/AzureBlobs/AzureStorageDevice.cs

```diff
@@ -292,8 +302,16 @@ public override void RemoveSegmentAsync(int segment, AsyncCallback callback, IAs
                     async (numAttempts) =>
                     {
                         var client = (numAttempts > 1) ? entry.PageBlob.Default : entry.PageBlob.Aggressive;
-                        await client.DeleteAsync(cancellationToken: this.PartitionErrorHandler.Token);
-                        return 1;
+                        try
+                        {
+                            using var response = await client.DeleteAsync(cancellationToken: this.PartitionErrorHandler.Token);
+                            return 1;
+                        }
+                        catch (Azure.RequestFailedException ex) when (numAttempts > 1 && BlobUtilsV12.BlobDoesNotExist(ex))
+                        {
+                            // blob may have already been deleted by the previous attempt
+                            return 0;
+                        }
```

Comment:
Could we use a `TryDeleteIfExists`-type API instead?

Reply:
Yes, that is a bit simpler. Will do.